x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java

```diff
@@ -174,6 +174,11 @@ public class EsqlFeatures implements FeatureSpecification {
      */
     public static final NodeFeature METRICS_SYNTAX = new NodeFeature("esql.metrics_syntax");
 
+    /**
+     * New internal resolve_fields API for ES|QL
+     */
+    public static final NodeFeature NEW_RESOLVE_FIELDS_API = new NodeFeature("esql.new_resolve_fields_api");
```

Comment:
nit: "new" in the names of things tends to age poorly, perhaps resolve_fields_for_esql?

Reply:
++ I pushed https://github.com/elastic/elasticsearch/pull/110738/commits/4ec1f8005510c9234a9986def3d635e34003321f